### PR TITLE
[Promotion] Remove all fluent interfaces

### PR DESCRIPTION
**Pull request by tuka217**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | no
| License       | MIT
| Doc PR        | no

### CODE REVIEW

File: src/Sylius/Component/Promotion/Model/Promotion.php [hunk 1]
```diff
@@ -416,8 +357,6 @@ public function removeRule(RuleInterface $rule)
     {
         $rule->setPromotion(null);
```
Comment (stloyd) [inline src/Sylius/Component/Promotion/Model/Promotion.php]: This should be removed as rule __must__ be assigned to promotion.

File: src/Sylius/Component/Promotion/Model/Promotion.php [hunk 2]
```diff
@@ -464,8 +401,6 @@ public function removeAction(ActionInterface $action)
     {
         $action->setPromotion(null);
```
Comment (stloyd) [inline src/Sylius/Component/Promotion/Model/Promotion.php]: This should be removed as action __must__ be assigned to promotion.

File: src/Sylius/Component/Promotion/Model/Promotion.php [hunk 3]
```diff
@@ -368,8 +313,6 @@ public function removeCoupon(CouponInterface $coupon)
     {
         $coupon->setPromotion(null);
```
Comment (stloyd) [inline src/Sylius/Component/Promotion/Model/Promotion.php]: This should be removed as coupon __must__ be assigned to promotion.

File: src/Sylius/Component/Promotion/Model/Promotion.php [hunk 4]
```diff
@@ -266,8 +221,6 @@ public function setUsed($used)
     public function incrementUsed()
     {
         $this->used++;
```
Comment (stloyd) [inline src/Sylius/Component/Promotion/Model/Promotion.php]: `++$this->used;`

File: src/Sylius/Component/Promotion/Model/Promotion.php (hunk 1 above)
Comment (michalmarcinkowski) [inline src/Sylius/Component/Promotion/Model/Promotion.php]: These changes are not related to this PR and that change should be discussed separately, same as all below.

**Comment by michalmarcinkowski** [comment]
Thank you Ania! :+1:

**Comment by rvanlaak** [comment]
Tried to figure out why fluent interfaces were removed. What did I miss? 

They only are allowed for builders now? http://docs.sylius.org/en/latest/bundles/SyliusProductBundle/product_builder.html

**Comment by lchrusciel** [comment]
@rvanlaak I don't feel to be an authority, so I will just link this [article](http://ocramius.github.io/blog/fluent-interfaces-are-evil/)

**Comment by rvanlaak** [comment]
That was a really nice read @lchrusciel , thanks! :+1: I'd conclude that builders because of their nature should be the only ones to use fluent interfaces.
